src/store.ts

```diff
@@ -17,7 +18,30 @@ export const toggleNavigation = (): StoreValue<typeof isNavigationOpen> => {
   return isNavigationOpen.get()
 }
 
+// fundraiser banner
 export const closeFundraiser = (): StoreValue<typeof isFundraiserOpen> => {
   isFundraiserOpen.set(false)
   return isFundraiserOpen.get()
 }
+
+// modal system utils
+export const openModal = (modalName: string): void => {
+  const clone = modalEntries.get().slice()
+
+  if (!clone.includes(modalName)) {
+    clone.push(modalName)
+    modalEntries.set(clone)
+  }
+}
+
+export const closeModal = (modalName: string): void => {
+  const clone = modalEntries.get().slice()
+
+  if (clone.includes(modalName)) {
+    modalEntries.set(clone.filter(el => el !== modalName))
+  }
+}
+
+export const isModalOpen = computed(modalEntries, (modals) => {
+  return (modalName: string) => modals.includes(modalName)
+})
```

Comment:
In the future, there might be cases where we need to open a modal from inside `.astro` file, rather than `.vue` files.
So implemented the modal-system of the project using `nanostore` package.